byteosaurus_hex.py

```diff
@@ -1933,7 +1942,388 @@ def vxlan():
+                for i in range(1, mpls_label_count):
+                    bos=0 #bottom_of_stack logic
+                    if i==mpls_label_count-1:
+                        bos=1
+                    mpls_headers = mpls_headers / MPLS(label=int(mpls_labels_list[i]), cos=0, s=bos, ttl=255)
+        else:
+            return None
+        #PW Codeword logic
+        if mpls_inputs[3] is True:
+            final_pkt = Ether(src=outer_src_mac, dst=outer_dst_mac, type=0x8847) / mpls_headers / EoMCW(zero=0, reserved=0, seq=0) / inner_pkt
+        else:
+            final_pkt = Ether(src=outer_src_mac, dst=outer_dst_mac, type=0x8847) / mpls_headers / inner_pkt
+        return final_pkt  
 
+#################################################################################################################
+def build_mpls(msg_type):
+    if msg_type == 'MPLS_ICMP':
+        # Get input parameters
+        mpls_input_param, common_param = requires("MPLS")
+        icmp_input_param, _ = requires('ICMP') 
+        inner_icmp_pkt, mpls_icmp_pkt = None, None
+        fuzzy = (input("Generate random MPLS ICMP Packet? (y/n) > ").strip()).lower()
+        if fuzzy == 'y':
+            # Build the inner ICMP packet
+            icmp_type = (input("ICMP Type (req/reply) > ").strip()).lower()
+            inputs = []
+            # Common parameters
+            for i in range(0, len(common_param)):
+                inputs.insert(i, input("{} > ".format(common_param[i])))
+            inner_icmp_pkt = icmp_packet(fuzzy, 'MPLS', icmp_type, inputs)
+            # Generate the Mpls packet
+            mpls_icmp_pkt = mpls_packet(fuzzy, inner_icmp_pkt, inputs) 
+            if inner_icmp_pkt != None and mpls_icmp_pkt != None:
+                logger.info("Inner ICMP packet built")
+                logger.info("MPLS ICMP Packet built")
+                mpls_icmp_pkt.show() 
+                return mpls_icmp_pkt, inputs[0], inputs[1]
+            else:
+                return None  
+        elif fuzzy == 'n':
+            labels = None
+            icmp_type = (input("ICMP Type (req/reply) > ").strip()).lower()
+            # Getting input parameters
+            inputs = []
+            del icmp_input_param[-1]  # Skipping VLAN tag for inner ICMP
+            for i in range(0, len(icmp_input_param)):
+                temp_input = input("Inner {} > ".format(icmp_input_param[i]))
+                inputs.insert(i, temp_input)
+            inner_icmp_pkt = icmp_packet(fuzzy, 'MPLS', icmp_type, inputs)
+            # Craft the MPLS packet. Get mpls input parameters
+            mpls_inputs = []
+            for i in range(0, len(mpls_input_param)):
+                temp_input = input("{} > ".format(mpls_input_param[i]))
+                if "Labels" in mpls_input_param[i]:
+                    #using a list comprehension with dummy variable x to create a list from the labels (which was a string)
+                    labels = [x.strip() for x in temp_input.split(',')]
```

Comment:
segment does not gracefully handle a user input of label value more than `1048575` as packet gets crafted down the line with label value 24. We need to handle this on our end, either by erroring out or popping off that label from the list.

later we also need to see if this is a likely bug on scapy side as the packet should not be generated.

Reply:
Addressed. Added the following -
```python
if (0 <= int(labels[k]) <= 1048575) is False:
   logger.error("Invalid input '{}' Expected label in range (0-1048575)".format(labels[k]))
```